pkg/guacanalytics/patchPlanning_test.go

```diff
@@ -0,0 +1,460 @@
+	logger := logging.FromContext(ctx)
+
+	topns := "topns"
+	topns1 := "topns1"
+	topns2 := "topns2"
+	topns3 := "topns3"
+	ns := "ubuntu"
+	version := "1.19.0"
+	depns := "openssl.org"
+	depns2 := "openssl.org2"
+	opensslVersion := "3.0.3"
+
+	ingestDependencies := []struct {
+		topTopPkg1           model.PkgInputSpec
+		topTopPkg2           model.PkgInputSpec
+		topTopPkg3           model.PkgInputSpec
+		topPkg               model.PkgInputSpec
+		pkg                  model.PkgInputSpec
+		depPkg1              model.PkgInputSpec
+		depPkg2              model.PkgInputSpec
+		dependencyDirect     model.IsDependencyInputSpec
+		dependencyNotInRange model.IsDependencyInputSpec
+		dependencyIndirect   model.IsDependencyInputSpec
+	}{{
+		topTopPkg1: model.PkgInputSpec{
+			Type:      "top1",
+			Namespace: &topns1,
+			Name:      "toppkg1",
+			Version:   &version,
+		},
+		topTopPkg2: model.PkgInputSpec{
+			Type:      "top2",
+			Namespace: &topns2,
+			Name:      "toppkg2",
+			Version:   &version,
+		},
+		topTopPkg3: model.PkgInputSpec{
+			Type:      "top3",
+			Namespace: &topns3,
+			Name:      "toppkg3",
+			Version:   &version,
+		},
+		topPkg: model.PkgInputSpec{
+			Type:      "top",
+			Namespace: &topns,
+			Name:      "toppkg",
+			Version:   &version,
+		},
+		pkg: model.PkgInputSpec{
+			Type:      "deb",
+			Namespace: &ns,
+			Name:      "dpkg",
+			Version:   &version,
+			Qualifiers: []model.PackageQualifierInputSpec{
+				{Key: "arch", Value: "amd64"},
+			},
+		},
+		depPkg1: model.PkgInputSpec{
+			Type:      "conan",
+			Namespace: &depns,
+			Name:      "openssl",
+			Version:   &opensslVersion,
+		},
+		depPkg2: model.PkgInputSpec{
+			Type:      "conan2",
+			Namespace: &depns2,
+			Name:      "openssl2",
+			Version:   &opensslVersion,
+		},
+		dependencyDirect: model.IsDependencyInputSpec{
+			VersionRange:   "=>1.19.0",
+			DependencyType: model.DependencyTypeDirect,
+			Justification:  "test justification one",
+			Origin:         "Demo ingestion",
+			Collector:      "Demo ingestion",
+		},
+		dependencyNotInRange: model.IsDependencyInputSpec{
+			VersionRange:   "<1.19.0",
+			DependencyType: model.DependencyTypeIndirect,
+			Justification:  "test justification one",
+			Origin:         "Demo ingestion",
+			Collector:      "Demo ingestion",
+		},
+		dependencyIndirect: model.IsDependencyInputSpec{
+			VersionRange:   "=>1.19.0",
+			DependencyType: model.DependencyTypeIndirect,
+			Justification:  "test justification one",
+			Origin:         "Demo ingestion",
+			Collector:      "Demo ingestion",
+		},
+	}}
+	for _, ingest := range ingestDependencies {
+		_, err1 := model.IsDependency(context.Background(), client, ingest.topPkg, ingest.pkg, ingest.dependencyDirect)
+		if err1 != nil {
+			logger.Errorf("Error in ingesting: %v\n", err1)
+		}
+		_, err2 := model.IsDependency(context.Background(), client, ingest.pkg, ingest.depPkg1, ingest.dependencyDirect)
```

Comment:
re-use err instead of redefining a new variable since it is no longer used after each err check.